src/xrpld/overlay/detail/OverlayImpl.cpp

```diff
@@ -1224,15 +1248,18 @@ OverlayImpl::relay(
     // We have more peers than the minimum (disabled + minimum enabled),
     // relay to all disabled and some randomly selected enabled that
     // do not have the transaction.
-    auto enabledTarget = app_.config().TX_REDUCE_RELAY_MIN_PEERS +
-        (total - minRelay) * app_.config().TX_RELAY_PERCENTAGE / 100;
+    auto const enabledTarget = relay  //
+        ? app_.config().TX_REDUCE_RELAY_MIN_PEERS +
+            (total - minRelay) * app_.config().TX_RELAY_PERCENTAGE / 100
+        : 0;
```

Comment:
You can remove the `relay` check from here, since it early returns above.
```suggestion
    auto const enabledTarget = app_.config().TX_REDUCE_RELAY_MIN_PEERS +
            (total - minRelay) * app_.config().TX_RELAY_PERCENTAGE / 100;
```
(This is the original code.)